♻️ ref: SLOs for `CommitContextIntegration`

iamrajjoshi opened the pull request:
observability for `CommitContextIntegration`, halts mainly encompass -  rate limiting or user config problems  (user disabling the bot, etc.)

iamrajjoshi committed:
:recycle: ref: lifecycle for get_blame_for_files

iamrajjoshi committed:
:recycle: ref: metrics for task and create_or_update_comment

iamrajjoshi committed:
:recycle: ref: use new interaction event class

iamrajjoshi committed:
can't use record_event

iamrajjoshi committed:
:wrench: chore: add some more record_failures

evanpurkhiser committed:
feat(crons): Link to docs for unknown check-in legend (#81012)

Requires https://github.com/getsentry/sentry-docs/pull/11894

iamrajjoshi commented on src/sentry/integrations/utils/metrics.py:
```diff
@@ -113,6 +113,10 @@ def add_extra(self, name: str, value: Any) -> None:
         """
         self._extra[name] = value
 
+    def add_extras(self, extras: Mapping[str, Any]) -> None:
```
quick little utility to add multiple extras at the same time. i was using this initially in this pr, but dropped the usage - still figured it might be worth keeping around

GabeVillalobos replied:
I've been thinking about this a bit. Do we want to make the typing on this stricter? Most of the time we're passing integers and strings around for logging only.
```suggestion
    def add_extras(self, extras: Mapping[str, int | str]) -> None:
```

ryan953 committed:
ci(eslint): Inline all the external eslint-config-sentry* ESLint configs into our .eslintrc file (#80970)

Related to https://github.com/getsentry/frontend-tsc/issues/81

I'm proposing that we inline all this eslint config into the repo,
instead of leveraging an external package. The idea is to make it easier
and faster to iterate on.

This PR is just an naive 7 mechanical operation: inline everything and
don't refactor too much. Each commit in this PR should be easy to
compare against https://github.com/getsentry/eslint-config-sentry with
some small things to note along the way.

The followup task will be
https://github.com/getsentry/frontend-tsc/issues/82 which is a good time
to squash objects and generally have a reason to refactor what's in
here. Possible strategies might be to stratify the config values based
on some/all of:
- plugin name
- runtime speed ie: `import slowRules from './eslint/slowRules.js';`
- etc.

ArthurKnaus committed:
feat(stats): Display total spans with stored (#81041)

Display the amount of stored spans if the shown span outcomes are
affected by dynamic sampling.

Closes https://github.com/getsentry/projects/issues/196
Closes https://github.com/getsentry/projects/issues/139

armenzg committed:
ref(releases): Function to handle resolved in release (#80983)

This is the largest code block in update_groups. This moves all related
logic into a single function.

oioki committed:
fix(ui): use marked superclass renderers for images and links (#81075)

Use marked superclass renderers instead of creating html manually.

Thank you @a2256014 for discovering the bug.

Zylphrex committed:
chore(explore): Make avg default aggregation (#81086)

narsaynorath committed:
ref(dashboards): Move widget type to parameter when adding to dashboard (#81068)

This modal is coupled strongly with discover saved queries. Moving this
up to a param so I could integrate it properly with Explore

Zylphrex committed:
fix(eap): Add missing transaction.op (#81110)

This was missing from the EAP definitions in both SnQL and RPC.

shellmayr committed:
feat(getting-started): add slow and fast example functions to profiling onboarding (#80886)

Zylphrex committed:
chore(explore): Add tooltip to disabled group by and sort by (#81090)

![Screenshot 2024-11-20 at 6 09
24 PM](https://github.com/user-attachments/assets/5784176a-2061-4a64-bb40-52acb3626be7)
![Screenshot 2024-11-20 at 6 09
36 PM](https://github.com/user-attachments/assets/231510da-bf49-4cd9-9d5e-4c73f70ce33f)

Zylphrex committed:
chore(eap): http.response_content_length should not be prefixed (#81120)

ryan953 committed:
fix(toolbar): Remove closing </body> tag so extra #login_as markup is not injected (#81106)

In getsentry we inject some login code to the bottom of rendered
templates, we don't need this to happen inside our toolbar popup; and we
can skip it by omitting the `</body>` tag form the template.

[`class
ViewAsRenderMiddleware`](https://github.com/getsentry/getsentry/blob/3cff20a7236510b918e785e3cd50a3cd20434207/getsentry/vendor/viewas/middleware.py#L102-L133)
will replace `</body>` with the content of
https://github.com/getsentry/getsentry/blob/3cff20a7236510b918e785e3cd50a3cd20434207/getsentry/templates/viewas/header.html#L1

ryan953 committed:
fix(toolbar): Force the correct, subdomained, url when navigating to the toolbar popup (#81107)

This is a followup to https://github.com/getsentry/sentry/pull/80003

In #80003 all the redirect and url stuff is still true... but we only do
the redirect hop at the end when the user was logged out to begin with.
If the user starts off logged in then there's no redirect on the server
side and the url that the toolbar asked for inside the popup
(`https://sentry.io/toolbar/$org/$project/login-success/`) will just
load.

What's going to happen now:
If the user **is not logged into** sentry already:
1. Open a popup with url =
`https://sentry.io/toolbar/$org/$project/login-success/`
2. User redirected into login flow, with `?redirect=....` appended
3. Login flow completes, user taken to
`https://$org.sentry.io/toolbar/$org/$project/login-success/` inside the
popup
4. Popup can communicate with the iframe ✅ 

If the user **is already** logged into sentry:
1. Open a popup with url =
`https://sentry.io/toolbar/$org/$project/login-success/`
2. `https://sentry.io/toolbar/$org/$project/login-success/` loads as
requested
3. NEW-> javascript redirect to
`https://$org.sentry.io/toolbar/$org/$project/login-success/` inside the
popup
4. Popup can communicate with the iframe ✅

nikkikapadia committed:
fix(issues): Text overflow on avatar hover card (#81123)

<!-- Describe your PR here. -->
I noticed that when the email is long on the hover card for the suspect
commit avatar on the issue details page, the text overflows and is
outside the hover card. I put in a `break-word` for the body text to fix
that.

Before

![issues-detail-before](https://github.com/user-attachments/assets/2ec0e1a9-ede4-41dd-b061-25b96e4764c6)

After

![issues-detail-after](https://github.com/user-attachments/assets/34b10d95-2bc6-4dfe-b3b0-d2e49efb307b)
<!--

  Sentry employees and contractors can delete or ignore the following.

-->

Zylphrex committed:
fix(explore): Force chart type when known (#81109)

In regular mode, the chart type can be inferred via the series name.
This doesn't work in top N mode when the series name are the table
values. In that case, we know there's only 1 type so force the chart
type.

matejminar committed:
fix(sandbox): Typo in the guide (#81125)

evanpurkhiser committed:
feat(uptime): Add "Allow Tracing" option to uptime form (#81081)

Looks like this

<img width="790" alt="image"
src="https://github.com/user-attachments/assets/98fa3d1a-473f-4bcd-aace-bb675faba8c2">

harshithadurai committed:
fix(dashboards): team rendering for Edit Access selector (#81059)

Problem:
The `useTeamsById()` hook only renders teams from the `TeamStore` which
causes the dashboard to error out when the Edit Access selector uses an
unavailable team.

Fix:
- Query the selected team by `teamId` to ensure it isn't undefined
- This PR also fixes the Search functionality by being able to search
for any team irrespective of `TeamStore`

---------

Co-authored-by: harshithadurai <harshi.durai@esentry.io>

Zylphrex committed:
feat(spans): Add sdk.version to snql spans (#81126)

This was missing in indexed spans snql eap but was present in rpc eap.

ryan953 committed:
fix(replay): Include a message that delte/merge is not supported for replay issues (#81122)

This is modeled after the performance config:
https://github.com/getsentry/sentry/blob/38e79dd514a71d24724ce5e8725e78f98983f4ff/static/app/utils/issueTypeConfig/performanceConfig.tsx#L9-L20

ideally both Perf and Replay issues would be deletable.... and the team
is checking on that. Until we get that confirmation tho... lets add the
message.

Zylphrex committed:
feat(explore): Use processed autocomplete names (#81008)

The processed autocomplete names will map things lcp to
`measurements.lcp` so we show them the way they used to be shown in
other parts of sentry. Additionally, we'll get units back by doing this.

narsaynorath committed:
feat(dashboards): Add to Dashboard from Explore (#81064)

Adds a button to the chart in explore so we can click "Add to Dashboard"
and use Explore as a jump off point for a dashboard widget.

The button has the following functionality:
- Takes the first 3 series from the chart and opens up the add to
dashboard modal
- The filter should transfer over
- When switching to table view, the table in dashboards gets filled with
the columns from the samples table if added from sample mode
- When added from aggregate mode, the table view gets the group bys
followed by the aggregates

Note: The group by doesn't currently get filled in for timeseries
charts. I will follow up with another PR to make group by work.

getsentry-bot committed:
Revert "feat(dashboards): Add to Dashboard from Explore (#81064)"

This reverts commit 9fb8494518e35a01dee3da68d0f43b177a4b90c0.

Co-authored-by: narsaynorath <22846452+narsaynorath@users.noreply.github.com>

natemoo-re committed:
feat(nav): add analytics tracking (#80512)

Adds `trackAnalytics` call for primary nav items on the new sidebar

asottile-sentry committed:
ref: upgrade pytest-rerunfailures (#81116)

this should allow us to upgrade pytest finally

<!-- Describe your PR here. -->

ryan953 committed:
feat(stories): Add RouteAnalyticsContextProvider to stories so we can see what is viewed (#81108)

I realized recently that we could have analytics about which story files
are most commonly referenced. It would help to know what's used often
because it might indicate something that needs more attention, similarly
something not referenced might be because it's either intuitive, or
unused in new code.

This is a half step though. This `RouteAnalyticsContextProvider` used
relies on the `react-hook:route-activated` hook provided to the
HookStore, in dev that won't be implemented. So the data we collect will
only be from people using sentry.io/stories which might not cover
everyone.

snigdhas committed:
fix(merged-issues): Update grouphash endpoint to use simple serializer (#81017)

Switching to a simple serializer for the grouphash endpoint. This
endpoint is used to load the merged-issues page on an issue and doesn't
need to gather all the extra info (stacktrace, packages, tags, etc)
about all events in order to populate the page.

Fixes https://github.com/getsentry/sentry/issues/80961

isabellaenriquez committed:
fix(billing): correct constants (#81073)

Correct strings for indexed span category.

ryan953 committed:
feat(hydration error): Create a tree-view that highlights dom mutations directly (#80808)

<img width="1251" alt="SCR-20241115-mtqw"
src="https://github.com/user-attachments/assets/a6aab141-87c2-4186-8568-89597e649cb9">

Fixes https://github.com/getsentry/sentry/issues/74358

roaga committed:
fix(autofix): Hide Open Autofix button on errors without a stacktrace (#81132)

Adds a check that hides Autofix in the Solutions Hub for errors with no
stacktrace.

scttcper committed:
feat(issues): Add activity drawer (#81025)

Commit:
feat(CI): Decrease frequency of docker compose scheduled runs to 1/day (#81138)

This decreases the frequency of these jobs for now, until we can get new
devservices working in CI

Zylphrex committed:
fix(explore): Add legends back in (#81143)

We want legends back in explore and truncate the tooltip as needed.

lobsterkatie committed:
feat(grouping): Collect grouphash metadata metrics (#81070)

This adds a number of metrics to track grouphash metadata:

- `grouping.grouphashmetadata.event_hash_basis` - tracks the overall grouping method, including whether or not a hybrid fingerprint was involved

- `grouping.grouphashmetadata.event_hashing_metadata.{hash_basis}` - for stacktrace, message, fingerprint, security report, and fallback hashing, tracks certain hashing-method-specific data (see `METRICS_TAGS_BY_HASH_BASIS` for a list of what's collected)

- `grouping.grouphashmetadata.create_or_update_grouphash_metadata` - times the process of collecting grouphash metadata

- `grouping.grouphashmetadata.backfill_needed` - tracks instances in which we'd backfill missing metadata, to help estimate the load that will create. As the accompanying comment says, this greatly overestimates the number of backfills we'd actually do (since we see the same grouphashes repeatedly, and each would only need to be backfilled once), but given that all we need is an upper bound in order to set our initial backfill sample rate, it's good enough for now.

To test the first two metrics, they and their tag values have been added to the grouphash metadata snapshots. (The timer metric wasn't included because there's not much to test there, and the backfill metric wasn't included because it's temporary.)

Commit:
feat(devservices): Add restart setting and use devservices external network (#81094)

add restart setting to automatically restart unless explicitly stopped
for postgres, and also make containers use devservices external network

asottile-sentry committed:
ref: disable branch coverage (#81140)

this disables branch coverage (coverage of jumps for loops / if
statements / lambdas / try-except / etc.) but drastically improves CI
time

we can revert this out once
https://github.com/nedbat/coveragepy/issues/1746 is resolved and we're
on a sufficiently new-enough version of coverage

<!-- Describe your PR here. -->

ameliahsu committed:
chore(api): make doc integration endpoints private (#81139)

make doc integration endpoints private

iamrajjoshi committed:
:wrench: chore(api): Make SentryAppsStatsEndpoint Private (#81141)

the endpoint is only used in admin tool

asottile-sentry committed:
ref: upgrade pytest (#81135)

<!-- Describe your PR here. -->

wmak committed:
fix(eap): handle span status (#81144)

- Span.status filters should not be special cased in eap cause we just
store the converted string instead of an enum

scttcper committed:
feat(issues): Add unmatching events to graph (#81099)

Commit:
feat(icon): add globe (#81148)

![Screenshot 2024-11-21 at 11 58
07 AM](https://github.com/user-attachments/assets/fc57860c-db04-4023-8f59-f36273d9809c)

narsaynorath committed:
feat(dashboards): Add to Dashboard from Explore (#81151)

Same as #81064 but with tests fixed.

jangjodi committed:
ref(similarity): Use project endpoint instead of feature flag in ui (#81103)

Use project details endpoint instead of feature flag in ui
Removing the use of the feature flag will allow us to stop serializing
it in project options endpoint, which is taking a long time

saponifi3d committed:
ref(workflow_engine): Updates to the data models (#80710)

## Description
As we continue to spec out the migrations for issue alerts / metric
alerts, we found a few more data model changes.

---------

Co-authored-by: Cathy Teng <cathy.teng@sentry.io>

Christinarlong committed:
chore(notifications): Fix typing for rest_framework/notification_action (#80974)

jennmueng committed:
feat(autofix): Write access is optional in check (#81100)

Makes sure the frontend treats the `githubWriteIntegration` as optional
and optionally passes `checkWriteAccess` to true

getsentry-bot committed:
Revert "Failsafe in case config for transaction_processing_store is incorrect (#80463)"

This reverts commit 59bf9d8b0d8620ea690cbb94eb0b5ce160e9cae1.

Co-authored-by: kneeyo1 <126607112+kneeyo1@users.noreply.github.com>

sentaur-athena committed:
fix(auth): get scope from request (#80569)

In my [previous PR](https://github.com/getsentry/sentry/pull/79740) I
assumed admin sets scopes for partner and we only use those for every
auth. Talking to the partner and following oauth standards, it makes
more sense if the partner asks for the scope they need in request params
and admin sets the maximum scope a partner can ask for.

michellewzhang committed:
feat(flags): add javascript onboarding (#81078)

adds javascript onboarding to the feature flag sidebar. also, if we
detect that `event.contexts.flags` is not undefined, we hide the first
half of the onboarding as specified in the designs

### updated copy 
<img width="394" alt="SCR-20241121-koun"
src="https://github.com/user-attachments/assets/0c690716-dfea-4b5b-9df6-b63e347de0eb">


### no integration yet 

<img width="395" alt="SCR-20241120-mhqa"
src="https://github.com/user-attachments/assets/5a832b85-9eb1-4496-ac29-a61693791f44">
<img width="394" alt="SCR-20241120-mhrb"
src="https://github.com/user-attachments/assets/fe9cd906-00b2-4276-81ba-8262689ab9a7">


https://github.com/user-attachments/assets/3fd3eade-aac4-4635-9fc9-ab5a71886651

### integration detected


https://github.com/user-attachments/assets/7c42a145-2a03-40ab-b066-279edeeed140

<img width="391" alt="SCR-20241120-mrni"
src="https://github.com/user-attachments/assets/09077c47-248b-4fd8-ad06-df293e9b5cdf">



- relates to
https://github.com/getsentry/team-replay/issues/502?reload=1?reload=1

### todo: 
- [x] analytics charts
- [ ] figure out CTA details (another way the sidebar can be triggered)

wedamija committed:
fix(migrations): Add newlines between each app in the lockfile (#81157)

When two adjacent lines in the lockfile are changed git/github detects a
merge conflict. An easy work around is to just add newlines in between
each application

<!-- Describe your PR here. -->

oioki committed:
fix(stats): hide stats if no project access (#81162)

Second attempt of https://github.com/getsentry/sentry/pull/80447

Comparing with the previous attempt, we now also allow project stats for
staff users (`or user.is_staff`).

The superuser case is already covered in this code:

https://github.com/getsentry/sentry/blob/2e5e0654c717adfa569be7875f44cb7e2bda4a74/src/sentry/api/serializers/models/project.py#L131-L136
`is_superuser` is True, therefore `has_access` is True, and the first
condition in this snippet works:
```
if attrs["has_access"] or user.is_staff:
```

szokeasaurusrex committed:
fix(debugfile): Recognize chunk-uploaded Proguard files (#81131)

The `detect_dif_from_path` function does not correctly identify Proguard
files which are chunk uploaded because these files have a
randomly-assigned temporary path, and the logic tries to guess whether
the file is a Proguard file based on its path. However, the
`detect_dif_from_path` function also takes an optional `name` option,
which for chunk-uploaded files, is the file name that is specified by
the client making the assemble call. This has not been a problem yet,
since Sentry CLI does not support chunk uploading Proguard files, but we
are adding support in
https://github.com/getsentry/sentry-cli/issues/2196, and so we need to
fix the backend to allow chunk uploads of Proguard files.

Here, we update the logic to check both the `path` and the `name` for
potentially matching a Proguard file.

iamrajjoshi committed:
chore(slack): Better Logging for Incident Alert Notifications (#81169)

harshithadurai committed:
fix(dashboards): Add and modify tooltips according to dashboard edit access (#81069)

- Added new tooltips to buttons that get disabled when user does not
have Edit Access
- Modified tooltip on Edit Access dropdown button when the menu is open
for a cleaner UI

---------

Co-authored-by: harshithadurai <harshi.durai@esentry.io>

jangjodi committed:
chore(similarity): Move feature flags to list not used in FE (#81104)

Add similarity feature flags to list of flags not used in the frontend
This will prevent them from being serialized in project index endpoint

---------

Co-authored-by: Josh Ferge <josh.ferge@sentry.io>

edwardgou-sentry committed:
feat(alerts): Update EAP Alert builder to use EAPSpanSearchQueryBuilder and Span Tags Provider (#81152)

Updates EAP Alert builder query filter field to use the
`EAPSpanSearchQueryBuilder` component, same as Explore.
Also uses the `SpanTagsProvider` and context.

armenzg committed:
ref(releases): Function to process a single group resolution (#81119)

iamrajjoshi committed:
Update src/sentry/integrations/utils/metrics.py

Co-authored-by: Gabe Villalobos <GabeVillalobos@users.noreply.github.com>
